### ISSUE TITLE
About information matrix

### ISSUE THREAD

**Issue by leilai125** [UditSinghParihar/g2oSLAM#2]
Hi, thanks for your sharing. I noticed that you set Information matrix for odometry edges,  = 500.0 0.0 0.0 500.0 0.0 500.0 and Information matrix for loop closure edges,  = 700.0 0.0 0.0 700.0 0.0 700.0. May I ask why you set information matrix in this way? I have some visual odometry data, but I don't have information matrix. I don't know how to set up. Thanks!

**Comment by UditSinghParihar** [comment]
The important thing to note here is that absolute value of information matrix doesn’t matter whether it’s 500 vs 700 or 5000 vs 7000. You just need to trust your loop closure edge more than your odometry edges. The range of absolute values though I took from ORB SLAM and RTABMAP, but you can change it to 5000 vs 7000 yourself.

**Comment by leilai125** [comment]
> The important thing to note here is that absolute value of information matrix doesn’t matter whether it’s 500 vs 700 or 5000 vs 7000. You just need to trust your loop closure edge more than your odometry edges. The range of absolute values though I took from ORB SLAM and RTABMAP, but you can change it to 5000 vs 7000 yourself.

Thanks! So it doesn't matter, as long as loop closure edge information matrix is larger.

Also, do you happen to have the implementation about detecting loop closure edges? Thank you very much!

**Comment by UditSinghParihar** [comment]
Yes, just ensure that loop closure edge has higher information matrix.

Here is the implementation of detection of loop closure and calculating transformation and doing pose graph slam.
https://github.com/UditSinghParihar/rord_slam

Apart from this you can use traditional bag of visual words or NETVLAD network for loop closure detection. And then use some feature matching algorithm for getting corespondences and ultimately transformation matrix.

**Comment by leilai125** [comment]
> Yes, just ensure that loop closure edge has higher information matrix.
> 
> Here is the implementation of detection of loop closure and calculating transformation and doing pose graph slam. https://github.com/UditSinghParihar/rord_slam
> 
> Apart from this you can use traditional bag of visual words or NETVLAD network for loop closure detection. And then use some feature matching algorithm for getting corespondences and ultimately transformation matrix.

Thank you very much! 

But I only find how theuy use loop_pairs.txt, I don't find how they generate loop_pairs.txt. Could you please refer me where I can find the loop clocure detection implemention?

Also, I only have monocular image sequence. Is this implementation compatible to my case? If not, could you refer me a good implementation that I can try? Thank you very much! Sorry, I am new to this area.

Really appreciate!!!

**Comment by UditSinghParihar** [comment]
Hi, we have just released the code for local feature matching in our RoRD GitHub repository and not loop closure detection. For loop closure detection, we were using RoRD feature matching and for loop closure detection, we were calculating local feature matching between the query image and all the images in the database set, which are a 50-meter range of the query image, and then the database image which has the maximum correspondences with the query image is used as loop closure pair. 
For this part, we haven't released the implementation.
I would suggest you use Bag of visual words or NetVLAD implementation for the same.
